source/Symbol/SwiftASTContext.cpp

```diff
@@ -6044,14 +6044,33 @@ SwiftASTContext::GetBitSize(lldb::opaque_compiler_type_t type,
   return 0;
 }
 
-uint64_t SwiftASTContext::GetByteStride(lldb::opaque_compiler_type_t type) {
-  if (type) {
-    const swift::irgen::FixedTypeInfo *fixed_type_info =
-        GetSwiftFixedTypeInfo(type);
-    if (fixed_type_info)
-      return fixed_type_info->getFixedStride().getValue();
+llvm::Optional<uint64_t>
+SwiftASTContext::GetByteStride(lldb::opaque_compiler_type_t type,
+                               ExecutionContextScope *exe_scope) {
+  if (!type)
+    return {};
+  swift::CanType swift_can_type(GetCanonicalSwiftType(type));
+  if (swift_can_type->hasTypeParameter()) {
+    if (!exe_scope)
+      return {};
+    ExecutionContext exe_ctx;
+    exe_scope->CalculateExecutionContext(exe_ctx);
+    auto swift_scratch_ctx_lock = SwiftASTContextLock(&exe_ctx);
+    CompilerType bound_type = BindAllArchetypes({this, type}, exe_scope);
+    // Note thay the bound type may be in a different AST context.
+    return bound_type.GetByteStride(exe_scope);
   }
-  return 0;
+
+  const swift::irgen::FixedTypeInfo *fixed_type_info =
```

Comment:
This looks like the cheapest of the three ways of getting the byte size.  Shouldn't we try it first?

Reply:
We always need to bind the archetypes, otherwise we'll go to IRGen and assert miserably.
Also, I'm not sure what's cheaper between going to IRGen or asking mirrors. Maybe the latter, but I haven't profiled either.

Reply:
Okay.